scripts/conf.py

```diff
@@ -229,63 +229,75 @@ def nodeset_dyn_lines(nodeset, lkp: Lookup = lkp):
     )
 
 
-def partitionlines(partition, lkp=lkp):
+def partitionlines(partition, lkp=lkp) -> str:
     """Make a partition line for the slurm.conf"""
-    part_name = partition.partition_name
-    lines = []
     MIN_MEM_PER_CPU = 100
-    defmem: int = MIN_MEM_PER_CPU
-    part_is_tpu = len(partition.partition_nodeset_tpu) > 0
 
-    def defmempercpu(template_link):
-        machine_conf = lkp.template_machine_conf(template_link)
-        return max(MIN_MEM_PER_CPU, machine_conf.memory // machine_conf.cpus)
+    def defmempercpu(nodeset: str) -> int:
+        template = lkp.cfg.nodeset.get(nodeset).instance_template
+        machine = lkp.template_machine_conf(template)
+        return max(MIN_MEM_PER_CPU, machine.memory // machine.cpus)
+
+    defmem = min(
+        map(defmempercpu, partition.partition_nodeset), default=MIN_MEM_PER_CPU
+    )
 
-    if len(partition.partition_nodeset) > 0:
-        defmem = min(
-            defmempercpu(lkp.cfg.nodeset.get(nodeset_name).instance_template)
-            for nodeset_name in partition.partition_nodeset
-        )
     nodesets = list(
         chain(
             partition.partition_nodeset,
             partition.partition_nodeset_dyn,
             partition.partition_nodeset_tpu,
         )
     )
+
+    is_tpu = len(partition.partition_nodeset_tpu) > 0
+    is_dyn = len(partition.partition_nodeset_dyn) > 0
+
+    oversub_exlusive = partition.enable_job_exclusive or is_tpu
+    power_down_on_idle = partition.enable_job_exclusive and not is_dyn
+
     line_elements = {
-        "PartitionName": part_name,
+        "PartitionName": partition.partition_name,
         "Nodes": ",".join(nodesets),
         "State": "UP",
         "DefMemPerCPU": defmem,
         "SuspendTime": 300,
-        "Oversubscribe": "Exclusive"
-        if partition.enable_job_exclusive or part_is_tpu
-        else None,
-        "PowerDownOnIdle": "YES" if partition.enable_job_exclusive else None,
+        "Oversubscribe": "Exclusive" if oversub_exlusive else None,
+        "PowerDownOnIdle": "YES" if power_down_on_idle else None,
         **partition.partition_conf,
     }
-    lines.extend([dict_to_conf(line_elements)])
 
-    return "\n".join(lines)
+    return dict_to_conf(line_elements)
```

Comment:
I've deleted `"\n".join(lines)` because `lines` does alway has a single element. 
I don't understand your comment, please resolve if you question got answered.

Reply:
I misread, makes sense.